src/components/TodoItem.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onDelete: (id: number) => void
+  updateTodo: (uTodo: Todo) => void,
+  loadingTodoIds: number[]
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  updateTodo,
+  loadingTodoIds,
+}) => {
+  const { id, completed, title } = todo;
+  const [isEdited, setIsEdited] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  const doubleClickHandler = (event: React.MouseEvent) => {
+    event.preventDefault();
+    setIsEdited(true);
+  };
+
+  const handleUpdate = (event: React.FormEvent) => {
+    event.preventDefault();
+    const updatedTodo = { ...todo, title: newTitle };
+
+    if (!newTitle) {
+      onDelete(id);
+    } else if (newTitle === title) {
+      setIsEdited(false);
+    } else {
+      updateTodo(updatedTodo);
+      setIsEdited(false);
+    }
+  };
+
+  const handlerChangeStatus = (e: React.ChangeEvent) => {
+    e.preventDefault();
```

Comment:
You don't need to preventDefault in the checkbox's onChange handler. It doesn't trigger a page reload.